eng/docker-compose/setup-connectors.ps1

```diff
@@ -41,70 +40,73 @@ catch {
     Write-Error "Please provide valid .env file."
 }
 
-$sourcePort=$envFile["CONNECT_SOURCE_PORT"]
-$sinkPort=$envFile["CONNECT_SINK_PORT"]
+$sourcePort = $envFile["CONNECT_SOURCE_PORT"]
+$sinkPort = $envFile["CONNECT_SINK_PORT"]
 
 $sourceBase = "http://localhost:$sourcePort/connectors"
 $sinkBase = "http://localhost:$sinkPort/connectors"
 $sourceUrl = "$sourceBase/postgresql-source"
 $sinkUrl = "$sinkBase/opensearch-sink"
 
- # Source connector
- if(IsReady($sourceBase))
- {
-     try {
-         $sourceResponse = Invoke-RestMethod -Uri $sourceUrl -Method Get
-         if($sourceResponse)
-         {
-             Write-Output "Deleting existing source connector configuration."
-             Invoke-RestMethod -Method Delete -uri $sourceUrl
-         }
-     }
-     catch {
+# Source connector
+if (IsReady($sourceBase)) {
+    try {
+        $sourceResponse = Invoke-RestMethod -Uri $sourceUrl -Method Get -SkipHttpErrorCheck
+
+        # only true if the response was 200
+        if ($null -ne $sourceResponse.name) {
```

Comment:
With more manual testing, I realized that the 404 response when the connector does not yet exist was throwing an error, rather than being captured in `$sourceResponse`. But it wasn't really _throwing an exception_... just logging an error. Which became problematic, as it was never caught and handled. So I found this way (`-SkipHttpErrorCheck`) to stop writing the error to `stderr` and instead let it be captured in a variable.